src/Microsoft.Azure.SignalR.Common/ServiceConnections/MultiEndpointServiceConnectionContainer.cs

```diff
@@ -416,6 +425,15 @@ private async Task WaitForClientsDisconnect(HubServiceEndpoint endpoint)
             Log.TimeoutWaitingClientsDisconnect(_logger, endpoint.ToString(), (int)_scaleTimeout.TotalSeconds);
         }
 
+        private IEnumerable<ServiceEndpoint> SingleOrNotSupported(IEnumerable<ServiceEndpoint> endpoints, ServiceMessage message)
+        {
+            if (endpoints.Count() == 1)
```

Comment:
what about 0?

Reply:
0 simply logs and in client invocation scenario our CI manager will wait for the result until timeout. So to avoid wasting time, suppose custom router should not return 0 endpoints. https://github.com/Azure/azure-signalr/blob/dev/src/Microsoft.Azure.SignalR.Common/ServiceConnections/MultiEndpointMessageWriter.cs#L136-L140

Any suggestion about check count under IEnumerable<T>? ToList() first?

Reply:
Separate 0/many in exceptions for better information.